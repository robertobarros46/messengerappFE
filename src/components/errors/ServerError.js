import React, { Component } from 'react';
import './ServerError.css';
import { Link } from 'react-router-dom';
import { Button } from 'antd';

class ServerError extends Component {
    render() {
        return (
            <div className="server-error-page">
                <h1 className="server-error-title">
                    500
                </h1>
                <div className="server-error-desc">
                    Oops! Algo de inesperado occoreu em nosso servidor. Tente voltar!
                </div>
                <Link to="/"><Button className="server-error-go-back-btn" type="primary" size="large">Go Back</Button></Link>
            </div>
        );
    }
}

export default ServerError;