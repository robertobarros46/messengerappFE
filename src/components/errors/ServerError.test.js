import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Link } from 'react-router-dom';
import { Button } from 'antd';

Enzyme.configure({ adapter: new Adapter() });

import ServerError from './ServerError';
describe('<ServerError />', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<ServerError />);
    });

    it('renders', () => {
        expect(wrapper.exists()).toBe(true);
    });

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should have one Link and one Button components', () => {
        expect(wrapper.find(Button, Link)).toHaveLength(1);
    });

    it('test whole content', () => {
        expect(wrapper.contains(
            <div className="server-error-page">
                <h1 className="server-error-title">
                    500
                </h1>
                <div className="server-error-desc">
                    Oops! Algo de inesperado occoreu em nosso servidor. Tente voltar!
                </div>
                <Link to="/"><Button className="server-error-go-back-btn" type="primary" size="large">Go Back</Button></Link>
            </div>
        )).toEqual(true);
    });
});