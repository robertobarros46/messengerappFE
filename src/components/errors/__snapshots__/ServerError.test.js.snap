// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ServerError /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ServerError />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="server-error-title"
        >
          500
        </h1>,
        <div
          className="server-error-desc"
        >
          Oops! Algo de inesperado occoreu em nosso servidor. Tente voltar!
        </div>,
        <Link
          replace={false}
          to="/"
        >
          <Button
            block={false}
            className="server-error-go-back-btn"
            ghost={false}
            loading={false}
            prefixCls="ant-btn"
            size="large"
            type="primary"
          >
            Go Back
          </Button>
        </Link>,
      ],
      "className": "server-error-page",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "500",
          "className": "server-error-title",
        },
        "ref": null,
        "rendered": "500",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Oops! Algo de inesperado occoreu em nosso servidor. Tente voltar!",
          "className": "server-error-desc",
        },
        "ref": null,
        "rendered": "Oops! Algo de inesperado occoreu em nosso servidor. Tente voltar!",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Button
            block={false}
            className="server-error-go-back-btn"
            ghost={false}
            loading={false}
            prefixCls="ant-btn"
            size="large"
            type="primary"
          >
            Go Back
          </Button>,
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "block": false,
            "children": "Go Back",
            "className": "server-error-go-back-btn",
            "ghost": false,
            "loading": false,
            "prefixCls": "ant-btn",
            "size": "large",
            "type": "primary",
          },
          "ref": null,
          "rendered": "Go Back",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="server-error-title"
          >
            500
          </h1>,
          <div
            className="server-error-desc"
          >
            Oops! Algo de inesperado occoreu em nosso servidor. Tente voltar!
          </div>,
          <Link
            replace={false}
            to="/"
          >
            <Button
              block={false}
              className="server-error-go-back-btn"
              ghost={false}
              loading={false}
              prefixCls="ant-btn"
              size="large"
              type="primary"
            >
              Go Back
            </Button>
          </Link>,
        ],
        "className": "server-error-page",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "500",
            "className": "server-error-title",
          },
          "ref": null,
          "rendered": "500",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Oops! Algo de inesperado occoreu em nosso servidor. Tente voltar!",
            "className": "server-error-desc",
          },
          "ref": null,
          "rendered": "Oops! Algo de inesperado occoreu em nosso servidor. Tente voltar!",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Button
              block={false}
              className="server-error-go-back-btn"
              ghost={false}
              loading={false}
              prefixCls="ant-btn"
              size="large"
              type="primary"
            >
              Go Back
            </Button>,
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "block": false,
              "children": "Go Back",
              "className": "server-error-go-back-btn",
              "ghost": false,
              "loading": false,
              "prefixCls": "ant-btn",
              "size": "large",
              "type": "primary",
            },
            "ref": null,
            "rendered": "Go Back",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
